radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -17,50 +17,100 @@
 
 package com.radixdlt.consensus;
 
+import com.google.common.collect.Maps;
 import com.google.inject.Inject;
 import com.radixdlt.consensus.validators.ValidationState;
 import com.radixdlt.consensus.validators.ValidatorSet;
 import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
 import com.radixdlt.crypto.Hash;
+import com.radixdlt.utils.Pair;
 
 import java.util.HashMap;
 import java.util.Objects;
 import java.util.Optional;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 /**
  * Manages pending votes for various vertices
  */
 public final class PendingVotes {
-	private final HashMap<Hash, ValidationState> votes = new HashMap<>();
+	private static final Logger log = LogManager.getLogger();
+
+	private final HashMap<View, Pair<Hash, ValidationState>> state = Maps.newHashMap();
 	private final Hasher hasher;
 
 	@Inject
 	public PendingVotes(Hasher hasher) {
 		this.hasher = Objects.requireNonNull(hasher);
 	}
 
+	/**
+	 * Prepare for a round of voting on the specified vote data by the specified validator set.
+	 *
+	 * @param voteData The data to be voted on
+	 * @param validatorSet The {@link ValidatorSet} that will be voting
+	 */
+	public void startVotingOn(VoteData voteData, ValidatorSet validatorSet) {
+		View voteView = voteData.getProposed().getView();
+		final Hash voteHash = hasher.hash(voteData);
+		this.state.put(voteView, Pair.of(voteHash,  validatorSet.newValidationState(voteHash)));
+	}
+
+	/**
+	 * Finish a round of voting for the specified view.
+	 *
+	 * @param view The view to finish voting for
+	 */
+	public void finishVotingFor(View view) {
+		this.state.remove(view);
+	}
+
 	/**
 	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.
 	 *
 	 * A QC will only be formed if permitted by the {@link ValidatorSet}.
 	 * @param vote The vote to be inserted
 	 * @return The generated QC, if any
 	 */
-	public Optional<QuorumCertificate> insertVote(Vote vote, ValidatorSet validatorSet) {
+	public Optional<QuorumCertificate> insertVote(Vote vote) {
+		View voteView = vote.getVoteData().getProposed().getView();
+		Pair<Hash, ValidationState> votingState = this.state.get(voteView);
+
+		final ECPublicKey voteAuthor = vote.getAuthor();
+		if (votingState == null) {
+			String replicaId = replicaId(voteAuthor);
+			log.warn("Ignoring invalid vote from {} for view {}", replicaId, voteView);

```

Comment:
It needn't necessarily be an invalid vote here, it could just be late.

Reply:
(depending on behaviour at line 107)

Reply:
Yeah, probably best just to change the wording to unexpected and make it an "info".